src/xcode/ENA/ENA/Source/Scenes/TicketValidation/CertificateSelection/TicketValidationCertificateSelectionViewModel.swift

```diff
@@ -0,0 +1,218 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import UIKit
+import OpenCombine
+import HealthCertificateToolkit
+
+class TicketValidationCertificateSelectionViewModel {
+
+	// MARK: - Init
+
+	init(
+		validationConditions: ValidationConditions,
+		healthCertificateService: HealthCertificateService,
+		onHealthCertificateCellTap: @escaping(HealthCertificate, HealthCertifiedPerson) -> Void
+	) {
+		self.validationConditions = validationConditions
+		self.healthCertificateService = healthCertificateService
+		self.onHealthCertificateCellTap = onHealthCertificateCellTap
+		
+		self.supportedHealthCertificates = []
+		self.filterCertificatesBasedOnValidationConditions()
+	}
+	
+	// MARK: - Internal
+
+	@OpenCombine.Published var dynamicTableViewModel: DynamicTableViewModel = DynamicTableViewModel([])
+	
+	// MARK: - Private
+	
+	private var healthCertificateService: HealthCertificateService
+	private var validationConditions: ValidationConditions
+	private let onHealthCertificateCellTap: (HealthCertificate, HealthCertifiedPerson) -> Void
+	private var supportedHealthCertificates: [HealthCertificate]
```

Comment:
I don't think you need to hold the certificates in a property, you could also just pass them into `dynamicTableViewModelSupportedHealthCertificates` just as you do with the person. I think that would be more consistent. Also makes it easier to extract the funcs that do the filtering